### DISCUSSION TITLE
GitHub Actions - No such file or directory for additional scripts

### DISCUSSION THREAD

**Discussion by asadmansr** [community/community#25197] [Actions]
<p>Hi,</p>
<p>I have created a GitHub Action (<a href="https://github.com/asadmansr/android-test-report-action" rel="nofollow noopener">https://github.com/asadmansr/android-test-report-action</a>) that uses a bash _entrypoint.sh _to execute a python script. However, when other projects use the action it fails by stating “python: can’t open file ‘extractReport.py’: [Errno 2] No such file or directory”.</p>
<p>I’m confused on why the python file is not being found even if I am copying it in the Dockerfile. How do I verify that the python file is there in the example project? It seems like entrypoint.sh is running, but the python file did not get carried over. Does that mean I cannot have additional scripts in my action?</p>
<p>Thanks.</p>
<p><strong>Android Test Report Action</strong></p>
<pre><code>COPY entrypoint.sh /entrypoint.sh
COPY extractReport.py /extractReport.py

ENTRYPOINT ["/entrypoint.sh"]
</code></pre>
<p>entrypoint.sh</p>
<pre><code>python extractReport.py "$i"
</code></pre>
<p><strong>Example Project</strong></p>
<pre><code>python: can't open file 'extractReport.py': [Errno 2] No such file or directory
</code></pre>
<p>Logs: <a href="https://github.com/asadmansr/android-test-report-action-example/runs/518101951?check_suite_focus=true" rel="nofollow noopener">https://github.com/asadmansr/android-test-report-action-example/runs/518101951?check_suite_focus=true</a></p>

**Comment by logankilpatrick** [comment]
<p>Hey <a class="mention" href="/asadmansr">@asadmansr</a>, I suggest using the “ls” command if you are running the action on macOS to verify that the file is being copied to where you think it is. </p>

**Comment by asadmansr** [comment]
<p>Hi <a class="mention" href="/logankilpatrick">@logankilpatrick</a> , thanks for reaching out.</p>
<p>Where should I run the “ls”? I ran it on the example project, but that only listed my project. I also built the Dockerfile locally, both entrypoint and python file are located in the root path ("/").</p>
<p>Does my python file in my Github Action have to be in a certain place within the docker image? I see a few paths here like (“github/workspace”, "github/home/).</p>
<pre><code>/usr/bin/docker run --name e87b52cb3a102a815d41bdbedf4d70f0634d5b_c742fa --label e87b52 --workdir /github/workspace --rm -e JAVA_HOME -e JAVA_HOME_8.0.242_x64 -e HOME -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/android-test-report-action-example/android-test-report-action-example":"/github/workspace" e87b52:cb3a102a815d41bdbedf4d70f0634d5b
</code></pre>

**Comment by logankilpatrick** [comment]
<p>Hey, yes the python file has to be in the current working directly when it is called. If you try to call the file and it is not in the current working directly, you get the error undefined issue that you are running into. </p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/asadmansr">@asadmansr</a> ,</p>
<p>In the action dockerfile, you can COPY the py script into a absolute path.</p>
<pre><code>COPY extractReport.py /usr/bin/extractReport.py
</code></pre>
<p>And in the entrypoint.sh, execute it.</p>
<pre><code>python /usr/bin/extractReport.py "$i"
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/0/02dcd6eb52b7178176678b2cca4841c8b02f305a.png" data-download-href="/uploads/short-url/pk6hXUgCWX4PvAqVUKxJsubhlU.png?dl=1" title="entrypoint.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181089350-e1208880-534e-4901-bc83-12bdd7163123.png" alt="entrypoint.png" data-base62-sha1="pk6hXUgCWX4PvAqVUKxJsubhlU" width="510" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/0/02dcd6eb52b7178176678b2cca4841c8b02f305a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">entrypoint.png</span><span class="informations">898×880 38.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by asadmansr** [comment]
<p>Thank you <a class="mention" href="/weide-zhou">@weide-zhou</a>! It’s working =)</p>
